tools/e2e-agent/disk.go

```diff
@@ -0,0 +1,190 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"math/rand"
+	"net/http"
+	"os"
+	"os/exec"
+	"strconv"
+	"time"
+
+	"k8s.io/klog/v2"
+)
+
+const (
+	// DiskImageSize is the default file size(1GB) used while creating backing image
+	DiskImageNamePrefix = "openebs-disk"
+)
+
+// LoopDevice has the attributes of a virtual disk which is emulated for testing
+type LoopDevice struct {
+	// Size in bytes
+	Size int64 `json:"size"`
+	// The backing image name
+	// eg: fake123
+	ImageName string `json:"imageName"`
+	// Image directory
+	// eg: /tmp
+	ImgDir string `json:"imgDir"`
+	// the disk name
+	// eg: /tmp/loop9002
+	DiskPath string `json:"diskPath"`
+}
+
+func (disk *LoopDevice) createDiskImage() error {
+	f, err := os.CreateTemp(disk.ImgDir, DiskImageNamePrefix+"-*.img")
+	if err != nil {
+		return fmt.Errorf("error creating disk image. Error : %v", err)
+	}
+	disk.ImageName = f.Name()
+	err = f.Truncate(disk.Size)
+	if err != nil {
+		return fmt.Errorf("error truncating disk image. Error : %v", err)
+	}
+
+	return nil
+}
+
+// Generates a random loop device name. The name will be of the
+// format loop9XXX, where X=[0-9]. The 4 digit numbering is chosen so that
+// we get enough disks to be randomly generated and also it does not clash
+// with the existing loop devices present in some systems.
+func getLoopDevName() string {
```

Comment:
Why not use `losetup -f` to get the next unused device name?

Reply:
https://github.com/openebs/openebs-e2e/compare/0ad2337317a29da2ddaedb95ca389b6d48e5e40a..0232e291a02224d4ceb55fe0d1dbd70388c7c935